src/openms/include/OpenMS/ANALYSIS/TOPDOWN/DeconvolutedSpectrum.h

```diff
@@ -0,0 +1,144 @@
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Kyowon Jeong, Jihyung Kim $
+// $Authors: Kyowon Jeong, Jihyung Kim $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/KERNEL/MSSpectrum.h>
+#include <OpenMS/ANALYSIS/TOPDOWN/FLASHDeconvHelperStructs.h>
+#include <OpenMS/ANALYSIS/TOPDOWN/PeakGroup.h>
+#include <iomanip>
+
+namespace OpenMS
+{
+  class PeakGroup;
+
+  /**
+       @brief A class representing a deconvoluted spectrum. Also contains deconvoluted precursro information for MSn n>1.
+  */
+  class OPENMS_DLLAPI DeconvolutedSpectrum :
+      public std::vector<PeakGroup>
+  {
+  public:
+    typedef FLASHDeconvHelperStructs::LogMzPeak LogMzPeak;
+
+    /// default constructor
+    DeconvolutedSpectrum() = default;
+
+    /**
+       @brief Constructor for DeconvolutedSpectrum
+       @param s spectrum
+       @param n scan number
+  */
+    explicit DeconvolutedSpectrum(const MSSpectrum &s, int n);
+
+    /// default deconstructor
+    ~DeconvolutedSpectrum() = default;
+
+    /// copy constructor
+    DeconvolutedSpectrum(const DeconvolutedSpectrum &) = default;
+
+    /// move constructor
+    DeconvolutedSpectrum(DeconvolutedSpectrum &&other) = default;
+
+    /// assignment operator
+    DeconvolutedSpectrum &operator=(const DeconvolutedSpectrum &fd) = default;
+
+    /**
+        @brief write the header in the output file (spectrum level)
+        @param fs file stream to the output file
+        @param n the index to the mass
+        @param detail if set true, detailed information of the mass (e.g., peak list for the mass) is written
+   */
+    static void writeDeconvolutedMassesHeader(std::fstream &fs, int &n, bool detail);
```

Comment:
check for every parameter if it can be const.. e.g. I would expect
static void writeDeconvolutedMassesHeader(std::fstream &fs, const int n, const bool detail);

Reply:
It is being revised..

Reply:
Oh it will take some time. Haha being revised.